### ISSUE TITLE
Unable to generate .mar files with custom requirements.txt in Docker GPU CUDA 11.1 image

### ISSUE THREAD

**Issue by AshwinRaikar88** [pytorch/serve#1212]
I've built the latest docker gpu image with cuda 11.1 for torch serve

When I try to register my custom model using the command
curl -X POST http://0.0.0.0:8081/models?url=/home/model-server/model-store/td.mar

I get this error:

{
  "code": 400,
  "type": "ModelException",
  "message": "Custom pip package installation failed for td"
}

The requirements.txt contains opencv-python as one of it's dependencies
It's manual installation is stuck and takes forever to install within the environment.

I want to deploy it on the AKS cluster but when I try to serve it on Kubernetes on AKS the prediction is stuck forever as cv2 was not imported in the custom handler script.

All help is appreciated, Thanks

**Comment by chauhang** [comment]
@AshwinRaikar88 Please add [install_py_dep_per_model=true](https://github.com/pytorch/serve/blob/master/docs/configuration.md#allow-model-specific-custom-python-packages) in config.properties if your model needs install package. Also verify that your requirements.txt file has all the required dependencies.

**Comment by AshwinRaikar88** [comment]
Yes @chauhang, I've tried all those steps in the .config files as well as verifying all the required files for requirements.txt.

One specific problem which I noticed while manually trying to install pip packages inside docker container was with Opencv python.

I had to build it by installing cmake package then run pip install opencv-python
Usually it takes a few mins to complete ......but the process took forever to complete, that's where even the mar file generation fails to complete with the error message "message": "Custom pip package installation failed for td"
I guess this has got something to do with the production build containers.
